src/almanack/metrics/metrics.yml

```
@@ -3,296 +3,354 @@ metrics:
   - name: "repo-path"
     id: "SGA-META-0001"
     result-type: "str"
+    direction: 0
     description: >-
       Repository path (local directory).
   - name: "repo-commits"
     id: "SGA-META-0002"
     result-type: "int"
+    direction: 1
     description: >-
       Total number of commits for the repository.
   - name: "repo-file-count"
     id: "SGA-META-0003"
     result-type: "int"
+    direction: 1
     description: >-
       Total number of files tracked within the repository.
   - name: "repo-commit-time-range"
     id: "SGA-META-0004"
     result-type: "tuple"
+    direction: 0
     description: >-
       Starting commit and most recent commit for the repository.
   - name: "repo-days-of-development"
     id: "SGA-META-0005"
     result-type: "int"
+    direction: 1
     description: >-
       Integer representing the number of days of development
       between most recent commit and first commit.
   - name: "repo-commits-per-day"
     id: "SGA-META-0006"
     result-type: "float"
+    direction: 1
     description: >-
       Floating point number which represents the number of commits
       per day (using days of development).
   - name: "almanack-table-datetime"
     id: "SGA-META-0007"
     result-type: "str"
+    direction: 0
     description: >-
       String representing the date when this table was generated
       in the format of '%Y-%m-%dT%H:%M:%S.%fZ', for example:
       2024-11-22T18:20:30.123456Z .
   - name: "almanack-version"
     id: "SGA-META-0008"
     result-type: "str"
+    direction: 0
     description: >-
       String representing the version of the almanack which was used
       to generate this table.
   - name: "repo-primary-language"
     id: "SGA-META-0009"
     result-type: "str"
+    direction: 0
     description: >-
       Detected primary programming language of the repository.
   - name: "repo-primary-license"
     id: "SGA-META-0010"
     result-type: "str"
+    direction: 0
     description: >-
       Detected primary license of the repository.
   - name: "repo-doi"
     id: "SGA-META-0011"
     result-type: "int"
+    direction: 0
     description: >-
       Repository DOI value detected from CITATION.cff file.
   - name: "repo-doi-publication-date"
     id: "SGA-META-0012"
-    result-type: "datetime"
+    result-type: "str"
+    direction: 0
     description: >-
       Repository DOI publication date detected from CITATION.cff file
       and OpenAlex.org.
+  - name: "repo-almanack-sustainability-score"
+    id: "SGA-META-0013"
+    result-type: "float"
+    direction: 0
+    description: >-
+      Score which summarizes other metrics as a single normalized value
+      to simplify understanding of software sustainability from the lens
+      of the Software Gardening Almanack.
```

```suggestion
      Score which summarizes other metrics as a single normalized value
      which represents how likely a repository will be maintained over time.
```

I'm not sure I like my suggestion, but it would be good to avoid using sustainability in the description.

I'm also a bit confused how we would use this metric in isolation of other repositories. Do you envision us providing any guidance on how to interpret these values? I imagine that any guidance we provide will change over time (given new metrics), and, we would need some empirical result to inform this guidance. (I'm sure you have plans for this!)

---

Thanks! Each metric in isolation with the combination of the `sustainability_correlation` (previously `direction`) are signs of sustained growth / sustainability (or lack thereof). I think we likely need to document especially anti-patterns where someone might want to improve their work, similar to what's displayed for linter rule docs (for example, [see here](https://docs.astral.sh/ruff/rules/complex-structure/)). Generally I think we could model this somehow by studying patterns from the existing wider ecosystems in open source, finding a pattern in repositories which we define as sustainable somehow. My instinct tells me that we might have to define the equal opposite of a `sustinability_score` to do this; that we'd have to define what makes a repository "dead" somehow.